test/e2e/suite.go

```diff
@@ -40,14 +43,29 @@ type EndToEndTestSuiteConfig struct {
 
 // Setup runs the setup function from the provider and other
 // procedures before running the test suite
-func (ts *EndToEndTestSuite) Setup() {
+func (ts *EndToEndTestSuite) Setup(t *testing.T) {
 	if err := ts.setup(); err != nil {
-		panic(err)
+		t.Fatal(err)
 	}
 
-	// Wait for the virtual kubelet (deployed as a pod) to become fully ready
-	if _, err := f.WaitUntilPodReady(f.Namespace, f.NodeName); err != nil {
-		panic(err)
+	// Wait for the virtual kubelet node resource to become fully ready
+	if err := f.WaitUntilNodeCondition(func(ev watch.Event) (bool, error) {
```

Comment:
Isn't there literallly a Kubernetes wait function to do this?

Reply:
Do you have a ref for this?

Reply:
More, like, could we use the wait primitives? https://godoc.org/k8s.io/apimachinery/pkg/util/wait

Reply:
@sargun So you don't want to use a watch here? Sorry I'm just confused as to what you are looking for specifically.

Reply:
Actually scratch that. For some reason, I thought the wait package had something where you could pass a conditionfunc to look at the results of a watch